packages/core/test/unit/query-generator/select-query.test.ts

```diff
@@ -50,22 +51,123 @@ describe('QueryGenerator#selectQuery', () => {
+      });
+    });
+
+    it('ignores 0 as offset with a limit', () => {
+      const sql = queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        offset: 0,
+        limit: 10,
+      }, User);
+
+      expectsql(sql, {
+        default: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] LIMIT 10;`,
+        mssql: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] OFFSET 0 ROWS FETCH NEXT 10 ROWS ONLY;`,
+        'db2 ibmi': `SELECT "id" FROM "Users" AS "User" ORDER BY "User"."id" FETCH NEXT 10 ROWS ONLY;`,
+      });
+    });
+
+    it('ignores 0 as offset without a limit', () => {
+      const sql = queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        offset: 0,
+      }, User);
+
+      expectsql(sql, {
+        default: `SELECT [id] FROM [Users] AS [User];`,
+      });
+    });
+
+    it('support 0 as limit', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        limit: 0,
+      }, User), {
+        default: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] LIMIT 0;`,
+        mssql: new Error(`LIMIT 0 is not supported by ${dialectName} dialect.`),
+        'db2 ibmi': `SELECT "id" FROM "Users" AS "User" ORDER BY "User"."id" FETCH NEXT 0 ROWS ONLY;`,
+      });
+    });
+
+    it('escapes limit', () => {
+      const sql = queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- testing invalid limit
+        limit: `';DELETE FROM user`,
+      }, User);
+
+      expectsql(sql, {
+        default: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] LIMIT ''';DELETE FROM user';`,
+        mssql: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] OFFSET 0 ROWS FETCH NEXT N''';DELETE FROM user' ROWS ONLY;`,
+        'db2 ibmi': `SELECT "id" FROM "Users" AS "User" ORDER BY "User"."id" FETCH NEXT ''';DELETE FROM user' ROWS ONLY;`,
+        'mariadb mysql': 'SELECT `id` FROM `Users` AS `User` ORDER BY `User`.`id` LIMIT \'\\\';DELETE FROM user\';',
+      });
+    });
+
+    it('escapes offset', () => {
+      const sql = queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        limit: 10,
+        // @ts-expect-error -- testing invalid offset
+        offset: `';DELETE FROM user`,
+      }, User);
+
+      expectsql(sql, {
+        default: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] LIMIT 10 OFFSET ''';DELETE FROM user';`,
+        mssql: `SELECT [id] FROM [Users] AS [User] ORDER BY [User].[id] OFFSET N''';DELETE FROM user' ROWS FETCH NEXT 10 ROWS ONLY;`,
+        'db2 ibmi': `SELECT "id" FROM "Users" AS "User" ORDER BY "User"."id" OFFSET ''';DELETE FROM user' ROWS FETCH NEXT 10 ROWS ONLY;`,
+        'mariadb mysql': 'SELECT `id` FROM `Users` AS `User` ORDER BY `User`.`id` LIMIT 10 OFFSET \'\\\';DELETE FROM user\';',
```

Comment:
```suggestion
```

Reply:
MariaDB and MySQL uses a backslash to escape the single quote.